addons/l10n_my_edi_extended/tests/test_new_submissions_flow.py

```diff
@@ -355,6 +355,87 @@ def test_12_multiple_moves_with_one_failed_submission(self):
         failed_invoice = self.submission_invoice.filtered(lambda inv: not inv.l10n_my_edi_state)
         self.assertEqual(len(failed_invoice), 1, 'One invoice has no state.')
 
+    def test_13_multiple_cron_runs(self):
+        """
+        Simulate the cron running more than once; ensure that we correctly update l10n_my_edi_retry_at for valid invoices.
+        For the purpose of the test, we will use two separate submissions.
+        """
+        all_invoices = self.env['account.move']
+        # First submission of 5 invoices
+        with patch(CONTACT_PROXY_METHOD, new=self._test_13_mock_first_submission):
+            first_batch = self.env['account.move']
+            for i in range(5):
+                first_batch |= self.init_invoice(
+                    'out_invoice', products=self.product_a, post=True,
+                )
+            with freeze_time('2024-07-15 10:00:00'):
+                first_batch.action_l10n_my_edi_send_invoice()
+
+        all_invoices |= first_batch
+
+        # Second submission of 5 invoices.
+        self.get_submission_status_count = 0
+        with patch(CONTACT_PROXY_METHOD, new=self._test_13_mock):
+            second_batch = self.basic_invoice
+            for i in range(4):
+                second_batch |= self.init_invoice(
+                    'out_invoice', products=self.product_a, post=True,
+                )
+            with freeze_time('2024-07-15 10:00:00'):
+                second_batch.action_l10n_my_edi_send_invoice()
+
+        all_invoices |= second_batch
+
+        self.get_submission_status_count = 0
+        with patch(CONTACT_PROXY_METHOD, new=self._test_13_mock):
+            with freeze_time('2024-07-15 10:00:00'):
+                self.get_submission_status_count += 1  # Done once during the sending flow
```

Comment:
I don't get the += 1 here, shouldn't it be done right before (or after, but not here) all the calls to `_cron_l10n_my_edi_synchronize_myinvois` instead of here and after the calls to the cron method?

Maybe
```python
def _run_cron(self):
    self.env['account.move']._cron_l10n_my_edi_synchronize_myinvois()
    self.cron_call_count += 1
```

Reply:
It's because the submission `action_l10n_my_edi_send_invoice` calls the `get_submission_statuses` endpoint as well.
So we want to increase the count once for that one as well, so to keep control over our mock function.

`submission_status_count = 0` => Call during submission
`submission_status_count = x > 0` => Calls from the cron